### PR TITLE
Change to using doubles in XR classes

### PR DESCRIPTION
**Pull request by BastiaanOlij**
Change to using doubles in our XR classes. Ran into some issues with GD Extensions but it seems like a sensible change anyway.

### CODE REVIEW

**Review by m4gr3d — CHANGES_REQUESTED** [review]

File: modules/mobile_vr/mobile_vr_interface.h [hunk 1]
```diff
@@ -59,14 +59,14 @@ class MobileVRInterface : public XRInterface {
 	float eye_height = 1.85;
```
[the remark this answers is not in the document]
Comment (BastiaanOlij) [reply modules/mobile_vr/mobile_vr_interface.h:59]: Thanks! missed that one :)

**Review by m4gr3d — APPROVED** [review]

**Comment by BastiaanOlij** [comment]
@reduz we think we fixed the issue in `godot-cpp` that breaks floats, but I think we should still make this change. Are you okay with me merging this?

**Review by akien-mga — APPROVED** [review]
Needs a rebase, otherwise seems good.

Not sure if we could find a convenient way to document when things are `float` or `double` by design, and when they can be configured with `real_t`.

**Comment by BastiaanOlij** [comment]
@akien-mga that's a difficult one. Doubles may be overkill in this situation but seeing we're mostly dealing with plugins and with GD extensions plugins convert everything to doubles anyway (well parameters and return values at least) there is little to be gained using floats.

I don't have any solid ideas here, but it seems to me we should be using real_t where there is a definite performance or memory benefit when using float so we can reserve the use of double for high end hardware.

**Comment by akien-mga** [comment]
Thanks!

**Comment by aaronfranke** [comment]
> Not sure if we could find a convenient way to document when things are `float` or `double` by design, and when they can be configured with `real_t`.

In terms of doing calculations for XR, `real_t` makes the most sense. However, if it's easier to bind to GDExtension using doubles then it makes sense to use doubles with those external APIs, especially if it ends up being converted to Variant anyway.

**Pull request merged**